src/test/kotlin/MicronautAuthenticationTest.kt

```diff
@@ -0,0 +1,221 @@
+        )
+    }
+
+
+    // global authentication tests
+    //@ParameterizedTest
+    //@MethodSource("testCases")
+
+    /*
+    fun `ensure that global authentication set to any authentication will add the required parameter`(testCasePath: String) {
+        val controller = setupTest(testCasePath)
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).anySatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_AUTHENTICATED"}
+        }
+
+        assertThat(functionParameter).anySatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+        }
+    }*/
+
+    @Test
+    fun `ensure that global authentication set to empty authentication will add the prohibited parameter`() {
+        val controller = setupTest("global_authentication_prohibited.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).anySatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).noneSatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+        }
+    }
+
+    @Test
+    fun `ensure that global authentication set to any AND empty authentication will add the optional parameter`() {
+        val controller = setupTest("global_authentication_optional.yml")
+        val functionAnnotations = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.annotations } }
+
+        val functionParameter = controller.flatMap { it.members }
+            .flatMap { (it as TypeSpec).funSpecs.flatMap { it.parameters } }
+
+        assertThat(functionAnnotations).anySatisfy{member ->
+            assertThat(member.className.canonicalName).isEqualTo("io.micronaut.security.annotation.Secured")
+            assertThat(member.members).anyMatch{ it.toString() == "SecurityRule.IS_AUTHENTICATED, SecurityRule.IS_ANONYMOUS"}
+        }
+
+        assertThat(functionParameter).anySatisfy{ member ->
+            assertThat(member.name).isEqualTo("authentication")
+            assertThat(member.type.isNullable == true)
```

Comment:
**nitpick:** No need for the `== true` here. In general, have a look at the warnings your IDE reports to you and try to fix them unless there is a reason not to.